docs/source/tutorials/knn_outlier.ipynb

```diff
@@ -0,0 +1,447 @@
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7b552a12",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Package installation (hidden on docs website).\n",
+    "# If running on Colab, may want to use GPU (select: Runtime > Change runtime type > Hardware accelerator > GPU)\n",
+    "\n",
+    "dependencies = [\"cleanlab\", \"matplotlib\", \"torch\", \"pylab\", \"keras\", \"sklearn\", \"timm\", ]\n",
+    "\n",
+    "if \"google.colab\" in str(get_ipython()):  # Check if it's running in Google Colab\n",
+    "    %pip install cleanlab  # for colab\n",
+    "    cmd = ' '.join([dep for dep in dependencies if dep != \"cleanlab\"])\n",
+    "    %pip install $cmd\n",
+    "else:\n",
+    "    missing_dependencies = []\n",
+    "    for dependency in dependencies:\n",
+    "        try:\n",
+    "            __import__(dependency)\n",
+    "        except ImportError:\n",
+    "            missing_dependencies.append(dependency)\n",
+    "\n",
+    "    if len(missing_dependencies) > 0:\n",
+    "        print(\"Missing required dependencies:\")\n",
+    "        print(*missing_dependencies, sep=\", \")\n",
+    "        print(\"\\nPlease install them before running the rest of this notebook.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f7e76f30",
+   "metadata": {},
+   "source": [
+    "Lets first set some seeds for reproducibility"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bb4de93d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import torch\n",
+    "import warnings\n",
+    "\n",
+    "SEED = 123\n",
+    "np.random.seed(SEED)\n",
+    "torch.manual_seed(SEED)\n",
+    "torch.backends.cudnn.deterministic = True\n",
+    "torch.backends.cudnn.benchmark = False\n",
+    "torch.cuda.manual_seed_all(SEED)\n",
+    "warnings.filterwarnings(\"ignore\", \"Lazy modules are a new feature.*\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f58d0618",
+   "metadata": {},
+   "source": [
+    "## 2. Fetch and scale the Cifar10 dataset\n",
+    "\n",
+    "Import `cifar10` dataset. After some basic preprocessing, we manually remove some categories from the training data thereby making them outliers in the test set. For this example we've chosen to remove all categories that are not an animal `[airplane, automobile, ship, truck]` from the training set `trainX`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "936e7629",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import matplotlib.pyplot as plt\n",
+    "from pylab import rcParams\n",
+    "\n",
+    "from keras.datasets import cifar10, cifar100\n",
+    "from sklearn.model_selection import train_test_split"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1ae35d12",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Load cifar10 datasets\n",
```

Comment:
Is it possible to get the datasets from torchvision instead of keras?  torchvision would be a more natural dependency for users to install with torch + timm, especially we can be more sure that future versions of torchvisions will work with existing/past/future versions of torch/timm.

No worries if it's a lot of work to change this, hopefully future versions of keras.datasets will play nice with various versions of torch & timm.  In the past, I've seen incompatibility issues with having multiple DL frameworks installed from different version.